action.yml

```diff
@@ -45,6 +49,14 @@ runs:
     - uses: reviewdog/action-setup@v1
       with:
         reviewdog_version: v0.20.2
+    - run: |
+        if [ -n "${{ inputs.fail_level }}" ]; then
+          echo "INPUT_FAIL_LEVEL=${{ inputs.fail_level }}" >> $GITHUB_ENV
+        elif [ "${{ inputs.fail_on_error }}" = 'true' ]; then
+            echo 'INPUT_FAIL_LEVEL=any' >> $GITHUB_ENV
+        else
+            echo 'INPUT_FAIL_LEVEL=none' >> $GITHUB_ENV
+        fi
```

Comment:
These changes introduce the potential for executing untrusted inputs. Given that there is no way to know how the `fail_level` and `fail_on_error` inputs are constructed in the calling workflow—or who might be able to influence them—it would be advisable to introduce an intermediate environment variable to mitigate the potential for arbitrary code injection. In addition, the `shell` property is [required when `run` is set](https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions#runsstepsshell):

```suggestion
    - run: |
         if [ -n "${INPUT_FAIL_LEVEL}" ]; then
           echo "INPUT_FAIL_LEVEL=${INPUT_FAIL_LEVEL}" >> "${GITHUB_ENV}"
         elif [ "${INPUT_FAIL_ON_ERROR}" = "true" ]; then
             echo "INPUT_FAIL_LEVEL=any" >> "${GITHUB_ENV}"
         else
             echo "INPUT_FAIL_LEVEL=none" >> "${GITHUB_ENV}"
         fi
      shell: bash
      env:
        INPUT_FAIL_LEVEL: ${{ inputs.fail_level }}
        INPUT_FAIL_ON_ERROR: ${{ inputs.fail_on_error }}
```

Reply:
@marcransome good points on avoiding untrusted inputs and defining the shell. Agree with your suggestions.